frappe/core/doctype/doctype/doctype.py

```diff
@@ -350,8 +350,10 @@ def setup_fields_to_fetch(self):
 
 		self.flags.update_fields_to_fetch_queries = []
 
-		if set(old_fields_to_fetch) != {df.fieldname for df in new_meta.get_fields_to_fetch()}:
-			for df in new_meta.get_fields_to_fetch():
+		new_fields_to_fetch = [df for df in new_meta.get_fields_to_fetch()]
```

Comment:
was this copy necessary? could've done this directly?

```suggestion
new_fields_to_fetch = new_meta.get_fields_to_fetch()
```

Reply:
Yep, thanks for pointing out. Missed it while testing some things locally.